java/src/main/native/src/ColumnViewJni.cu

```diff
@@ -51,4 +54,18 @@ new_column_with_boolean_column_as_validity(cudf::column_view const &exemplar,
   return deep_copy;
 }
 
+std::unique_ptr<cudf::column> generate_list_offsets(cudf::column_view const &list_length,
+                                                    rmm::cuda_stream_view stream) {
+  CUDF_EXPECTS(list_length.type().id() == cudf::type_id::INT32,
+               "Input column does not have type INT32.");
+
+  auto offsets = cudf::make_numeric_column(cudf::data_type{cudf::type_id::INT32},
+                                           list_length.size() + 1, cudf::mask_state::UNALLOCATED);
+  auto output_view = offsets->mutable_view();
+  thrust::inclusive_scan(rmm::exec_policy(stream), list_length.template begin<offset_type>(),
+                         list_length.template end<offset_type>(),
+                         output_view.template begin<offset_type>() + 1);
```

Comment:
This will cause undefined behavior, since it will leave the first offset value uninitialized.

Reply:
In addition, the entire function `generate_lists_offsets` has been implemented in cudf. It is in the `include/cudf/string/detail/utilities.cuh`. We can try to expose it to be accessed here.

Reply:
See: https://github.com/rapidsai/cudf/blob/branch-22.06/cpp/include/cudf/strings/detail/utilities.cuh#L52

Reply:
I replaced this implementation with a call of `make_offsets_child_column`.